rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/internal_interpreter.kt

```diff
@@ -859,36 +991,48 @@ class InternalInterpreter(val systemInterface: SystemInterface) : InterpreterCor
             value2 is BlanksValue && value1 is StringValue -> value1.isBlank()
 
             value1 is StringValue && value2 is StringValue -> {
-                val v1 = value1.value.trimEnd().removeNullChars().trimEnd()
-                val v2 = value2.value.trimEnd().removeNullChars().trimEnd()
+                val v1 = value1.value.trimEnd()
+                val v2 = value2.value.trimEnd()
                 v1 == v2
             }
 
             value1 is DataStructValue && value2 is StringValue -> {
-                val v1 = value1.value.trimEnd().removeNullChars().trimEnd()
-                val v2 = value2.value.trimEnd().removeNullChars().trimEnd()
+                val v1 = value1.asStringValue().trimEnd()
+                val v2 = value2.value.trimEnd()
                 v1 == v2
             }
             value1 is StringValue && value2 is DataStructValue -> {
-                val v1 = value1.value.trimEnd().removeNullChars().trimEnd()
-                val v2 = value2.value.trimEnd().removeNullChars().trimEnd()
+                val v1 = value1.value.trimEnd()
+                val v2 = value2.asStringValue().trimEnd()
                 v1 == v2
             }
+            // To be review
```

Comment:
We should remove this comment.
Also, why a specific behavior for ProjectedArrayValue and not for ArrayValue?
Why getting element 1?